### ISSUE TITLE
DNS Resolve issue in Google Container Engine

### ISSUE THREAD

**Issue by nickveenhof** [rimusz/glusterfs-gce#8]
See https://github.com/kubernetes/kubernetes/issues/40180 for details. Looks like the example in this github repository isn't working as expected, but it is related to a bug somewhere in kubernetes and/or Google Cloud.

**Comment by simontesar** [comment]
I just want to note that mounting NFS volumes via a server that is referenced as ExternalName Service also fails for me in a Kubernetes Cluster installed on premises.

**Comment by rimusz** [comment]
Thanks guys for reporting, but this is Kubernetes bug not this project ones :)

But let's keep this issue opened.

**Comment by nickveenhof** [comment]
True, curious if we could adapt the example in this repository so that it works on Google Container Engine?

**Comment by rimusz** [comment]
it used to work. I'm not using gluster myself, this one was just an example

**Comment by Locus99** [comment]
I have gotten glusterfs to work on GCE in a kubernetes cluster.  To do this, I had to do 2 things:

1. Add a storageClassName tag to both the pv and pvc yaml files in the spec section.  If you don't do this (as far as I can tell), you will actually mount a different pv provided by kubernetes instead.  In other words, it will appear to work just fine, but you are not actually using the gluster mount.  The actual value in the storageClassName doesn't matter, but they need to match between the two yaml files.
2. I had to switch from using hostnames in the volume create command to IP addresses.  This is a fairly trivial change in the create_volume.sh file:
`CLUSTER=$CLUSTER${SPACE}${STATIC_IP[$i-1]}:/data/brick1/${VOLUME}`
but is more complicated in the create_cluster.sh script as the "gluster peer probe ${SERVER}-${i}" is not sufficient since you will need to also do a peer probe with the IP address as well.  I did this manually as peer probing with an IP address on the same host will fail (doesn't allow localhost).  So I had to log into different hosts to peer probe with that IP.  If I update the script at a later date to address this issue, I will post it here.

Note that none of these changes are required to enable glusterfs to work outside a kubernetes cluster, but are necessary at this time to work with kubernetes (at least with gce).

**Comment by rimusz** [comment]
thanks @Locus99 for the info 👍

**Comment by poor-bob** [comment]
Would be very exciting to see this :) 👍

**Comment by rimusz** [comment]
this is a nice approach to get PVCs automatically provisioned, should solve the dns issue
https://github.com/kubernetes-incubator/external-storage/tree/master/gluster/glusterfs
https://github.com/kubernetes-incubator/external-storage/blob/master/gluster/glusterfs/deploy/storageclass.yaml
